source/read/change-streams.txt

```diff
@@ -0,0 +1,272 @@
+
+   To learn how to create a free MongoDB Atlas cluster and load the sample datasets,
+   see the :atlas:`Get Started with Atlas </getting-started>` guide.
+
+Some examples use instances of the ``LatchedObserver`` class to handle change
+stream events. This class is a custom observer that prints change stream events
+and continues monitoring data changes until the stream completes or generates
+an error. To use the ``LatchedObserver`` class, paste the following code into
+your application file:
+
+.. literalinclude:: /includes/read/change-streams.scala
+    :language: scala
+    :dedent:
+    :start-after: start-latched-observer
+    :end-before: end-latched-observer
+
+Open a Change Stream
+--------------------
+
+To open a change stream, call the ``watch()`` method. The instance on which you
+call the ``watch()`` method determines the scope of events that the change
+stream monitors. You can call the ``watch()`` method on instances of the following
+classes:
+
+- ``MongoClient``: Monitor all changes in the MongoDB deployment
+- ``MongoDatabase``: Monitor changes in all collections in the database
+- ``MongoCollection``: Monitor changes in the collection
+
+The following example calls the ``watch()`` method to open a change stream on the
+``restaurants`` collection. The code creates a ``LatchedObserver`` instance to
+receive and output changes as they occur:
+
+.. literalinclude:: /includes/read/change-streams.scala
+   :start-after: start-open-change-stream
+   :end-before: end-open-change-stream
+   :language: scala
+   :dedent:
+
+To begin watching for changes, run the preceding code. Then, in a separate
+shell, modify the ``restaurants`` collection. The following example updates
+a document that has a ``name`` field value of ``"Blarney Castle"``:
+
+.. _scala-change-stream-update:
+
+.. literalinclude:: /includes/read/change-streams.scala
+   :start-after: start-update-for-change-stream
+   :end-before: end-update-for-change-stream
+   :language: scala
+   :dedent:
+
+When you update the collection, the change stream application prints the change
+as it occurs. The printed change event resembles the following output:
+
+.. code-block:: none
+   :copyable: false
+
+   ChangeStreamDocument{ operationType=update, resumeToken={"_data": "..."},
+   namespace=sample_restaurants.restaurants, destinationNamespace=null, 
+   fullDocument=null, fullDocumentBeforeChange=null, documentKey={"_id": {...}},
+   clusterTime=Timestamp{...}, updateDescription=UpdateDescription{removedFields=[],
+   updatedFields={"cuisine": "Irish"}, truncatedArrays=[], disambiguatedPaths=null},
+   txnNumber=null, lsid=null, splitEvent=null, wallTime=BsonDateTime{...}}
+
+Modify the Change Stream Output
+-------------------------------
+
+To modify the change stream output, you can pass a list of pipeline stages as a
+parameter to the ``watch()`` method. You can include the following stages in the
+list:
+
+- ``$addFields`` or ``$set``: Adds new fields to documents
```

Comment:
S: somewhere in this section, include a link to Server docs that describe these stages in more detail

https://www.mongodb.com/docs/manual/reference/operator/aggregation-pipeline/